File: src/main/java/macro303/keschet/pieces/Thief.java
package macro303.keschet.pieces;

import macro303.keschet.ConsoleColour;
import macro303.keschet.Direction;

public class Thief extends Piece {

	public Thief(ConsoleColour teamColour) {
		super(teamColour, 1, "T", new Direction[]{Direction.NORTH, Direction.NORTH_EAST, Direction.EAST, Direction.SOUTH_EAST, Direction.SOUTH, Direction.SOUTH_WEST, Direction.WEST, Direction.NORTH_WEST});
	}

	@Override
	public String toString() {
		return "Thief{} " + super.toString();
	}
}

File: Base/src/main/kotlin/macro303/base/pieces/Thief.kt
package macro303.base.pieces

import macro303.base.Colour
import macro303.base.Direction

class Thief(teamColour: Colour) : Piece(
	teamColour = teamColour,
	maxDistance = 1,
	symbol = "T",
	validDirections = arrayOf(
		Direction.NORTH,
		Direction.NORTH_EAST,
		Direction.EAST,
		Direction.SOUTH_EAST,
		Direction.SOUTH,
		Direction.SOUTH_WEST,
		Direction.WEST,
		Direction.NORTH_WEST
	)
) {

	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other !is Thief) return false
		if (!super.equals(other)) return false
		return true
	}

	override fun hashCode(): Int {
		return super.hashCode()
	}

	override fun toString(): String {
		return "Thief() ${super.toString()}"
	}
}